core/templates/pages/library-page/library-page.component.spec.ts

```diff
@@ -16,532 +16,559 @@
-          }, {
-            activity_type: 'invalid',
-            id: '3',
-            title: 'Invalid Summary'
-          }, {
-            activity_type: 'exploration',
-            id: 'exp4',
-            title: 'Exploration Summary'
-          }]
-        }]
-      });
-      $httpBackend.expectGET('/creatordashboardhandler/data').respond({
-        explorations_list: [{id: 'exp2'}, {id: 'exp3'}],
-        collections_list: [{id: 'col2'}, {id: 'col3'}]
-      });
+  it('should initialize', fakeAsync(() => {
+    spyOn(loaderService, 'showLoadingScreen');
+    spyOn(urlInterpolationService, 'getStaticImageUrl');
+    spyOn(
+      classroomBackendApiService, 'fetchClassroomPromosAreEnabledStatusAsync')
+      .and.returnValue(Promise.resolve(true));
+    spyOn(pageTitleService, 'setPageTitle');
+    spyOn(libraryPageBackendApiService, 'fetchLibraryGroupDataAsync')
+      .and.returnValue(Promise.resolve({
+        activity_list: [],
+        header_i18n_id: '',
+        preferred_language_codes: []
+      }));
+    spyOn(i18nLanguageCodeService.onPreferredLanguageCodesLoaded, 'emit');
+    spyOn(loaderService, 'hideLoadingScreen');
+    spyOn(windowDimensionsService, 'getWidth').and.returnValue(900);
+    componentInstance.ngOnInit();
+    spyOn(componentInstance, 'initCarousels');
+    tick();
+    tick();
+    expect(loaderService.showLoadingScreen).toHaveBeenCalledWith(
+      'I18N_LIBRARY_LOADING');
+    expect(classroomBackendApiService.fetchClassroomPromosAreEnabledStatusAsync)
+      .toHaveBeenCalled();
+    expect(componentInstance.CLASSROOM_PROMOS_ARE_ENABLED).toBeTrue();
+    expect(pageTitleService.setPageTitle).toHaveBeenCalled();
+    expect(libraryPageBackendApiService.fetchLibraryGroupDataAsync)
+      .toHaveBeenCalled();
+    expect(i18nLanguageCodeService.onPreferredLanguageCodesLoaded.emit)
+      .toHaveBeenCalled();
+    expect(loaderService.hideLoadingScreen).toHaveBeenCalled();
+    expect(windowDimensionsService.getWidth).toHaveBeenCalled();
+  }));
 
-      $scope = $rootScope.$new();
-      ctrl = $componentController('libraryPage', {
-        $scope: $scope,
-        I18nLanguageCodeService: i18nLanguageCodeService
-      });
+  it('should initialize for non group pages', fakeAsync(() => {
+    spyOn(loaderService, 'showLoadingScreen');
+    spyOn(urlInterpolationService, 'getStaticImageUrl');
+    spyOn(
+      classroomBackendApiService, 'fetchClassroomPromosAreEnabledStatusAsync')
+      .and.returnValue(Promise.resolve(true));
+    spyOn(pageTitleService, 'setPageTitle');
+    windowRef.nativeWindow.location.pathname = '/community-library';
+    fixture.detectChanges();
+    spyOn(libraryPageBackendApiService, 'fetchLibraryIndexDataAsync')
+      .and.returnValue(Promise.resolve(libraryIndexData as LibraryIndexData));
+    spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+      new UserInfo(
+        'role', true, true, true, true, true, 'en', 'user',
```

Comment:
Hey @ashutoshc8101, this change is causing type checks to fail since the `roles` parameter expects a `string[]` and you've passed in a `string`. Can you please check this?

Reply:
This happened because Sandeep PR was just merged 1 hr before my PR and my PR was not updated with the latest changes.

Reply:
Ok, thanks for updating 👍